llama_index/prompts/mixin.py

```diff
@@ -0,0 +1,90 @@
+"""Prompt Mixin."""
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import Dict, Optional, Tuple, Union
+
+from llama_index.prompts.base import BasePromptTemplate
+from copy import deepcopy
+
+HasPromptType = Union["PromptMixin", BasePromptTemplate]
+PromptDictType = Dict[str, BasePromptTemplate]
+PromptMixinType = Dict[str, "PromptMixin"]
+
+
+class PromptMixin(ABC):
+    """Prompt mixin.
+
+    This mixin is used in other modules, like query engines, response synthesizers.
+    This shows that the module supports getting, setting prompts,
+    both within the immediate module as well as child modules.
+
+    """
+
+    def _validate_prompts(
+        self, prompts_dict: PromptDictType, module_dict: PromptMixinType
+    ) -> None:
+        """Validate prompts."""
+        # validate each prompt module to make sure there's no overlapping keys
+        # (otherwise update will fail)
+        # start a counter of keys
+        all_prompt_keys = defaultdict(int)
+        # add keys from prompt dict
+        for key in prompts_dict:
+            all_prompt_keys[key] += 1
+        # add keys from each prompt module
+        for module in module_dict.values():
+            for key in module.get_prompts():
+                all_prompt_keys[key] += 1
+        # check for duplicates
+        for key, count in all_prompt_keys.items():
+            if count > 1:
+                raise ValueError(f"Duplicate prompt key {key} found.")
```

Comment:
is this still relevant? I think it's kinda weird that `get_prompts` can give value error.

if we really want to solve the duplicate key problem, we should just namespace the key e.g. `<module_name>-<prompt_key>`

Reply:
seems like we should still validate against duplicate keys somehow to prevent unexpected behavior? unless you disagree

agree namespacing is one solution, but also seems more complex. can look into it